test/GuzzleHttpServiceTest.php

```diff
@@ -99,17 +99,19 @@ public function testCanSendJsonRequests()
         $request = $r->getValue($diagnostic);
 
         if ($request instanceof RequestInterface) {
-            $this->assertSame('application/json', $request->getHeaderLine('Content-Type'));
+            self::assertSame('application/json', $request->getHeaderLine('Content-Type'));
         } else {
-            $this->assertSame('application/json', $request->getHeader('Content-Type'));
+            self::assertSame('application/json', $request->getHeader('Content-Type'));
         }
 
         $body = (string) $request->getBody();
-        $this->assertSame(['foo' => 'bar'], json_decode($body, true));
+        self::assertSame(['foo' => 'bar'], json_decode($body, true));
     }
 
-    public function testCanSendArbitraryRequests()
+    public function testCanSendArbitraryRequests(): void
     {
+        self::markTestSkipped('Clarify what to do with assertion of protected property.');
```

Comment:
To be discussed **before** merging

Reply:
Yeah, just use reflection for now as it might test something which has been fixed before.
I dont really like checking protected/private properties at all but this is done in so many packages and I dont get the reason behind that...

Reply:
Urgh, this went under my radar: OK to merge without it